docs/proposals/mcs/remove-derived-mcs.md

```diff
@@ -0,0 +1,352 @@
+
+The workflow of the additional dnsvserver as follows:
+
+![](.\additional-dns-server.jpg)
+
+The advantages and disadvantages of this method are as follows:
+
+advantage:
+
+1. Load balancing can be achieved. Since the extended dns resolver is customized, customers can achieve load balancing here.
+
+2. Access in the original way can be realized. Rules for local services and remote services can be determined in custom plug-ins. Therefore, the original access method can be maintained, in this case is  `foo.default.svc.cluster.local`.
+
+3. No other configuration of dns is required. Since there is a custom dns resolver, there is no need to configure dns search. Therefore, there is no configuration of dns search in solution one.
+
+shortcoming:
+
+1. The coredns component needs to be modified. It is also necessary to recompile coredns in order to install the corresponding plug-ins, and also need to modify the corresponding coredns configuration.
+
+2. New components need to be deployed in member clusters. This solution requires a dns extension resolver to be installed.
+
+### 2) Karmada adaptation
+
+What needs to be done in the `karmada` part is to fill in the `ips` field of the `ServiceImport` which imported to the member cluster on the existing basis and correspond to the `clusterIP` of `derived-foo`. Its flow diagram is as follows:
+
+![remove-derived-in-controller](./remove-derived-in-controller.jpg)
+
+As shown in the figure above, we still take the foo service as an example. After configuring coredns, we add the logic to obtain the derived-foo service in the cluster members in the ServiceImport-controller of the control plane.
+
+1. Wait for cluster member config the `derived-foo` service's `clusterIP` .
+
+2. The controller of the control plane obtains the value, and fills the value into the work corresponding to the `ServiceImport` of the control plane.
+
+3. According to the synchronization principle of `Karmada`, the `ServiceImport` object in the member cluster will be filled with the `clusterIP` value of `derived-foo`.
+
+That is, the effect of `ServiceImport` after completion needs to be:
+
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ServiceImport
+metadata:
+   name: foo
+   namespace: default
+spec:
+   type: ClusterSetIP
+   ports:
+   - name: "http"
+     port: 80
+     protocol: TCP
+   ips:
+   - "10.10.0.5" # clusterIP for cross-cluster access
+```
+
+### 3) Suggested operation
+
+The recommended configuration instructions for the `coreDNS`+`multicluster` plug-in solution are given below.
+
+#### Environment preparation:
+
+A normal Karmada cluster, and can perform existing mcs operations.
+
+The following uses the cluster member2 where the client resides as an example to access the `foo`service using the suffix `clusterset.local`. This solution needs to be configured as follows:
+
+#### coredns compile
+
+Refer to [Compile Instructions](#Compilation Instructions).
```

Comment:
Expired Link

Reply:
This is the difference between github and typora. Let me find a solution